core/cog-launcher.c

```diff
@@ -638,3 +669,51 @@ cog_launcher_add_web_cookies_option_entries (CogLauncher *launcher)
     g_application_add_option_group (G_APPLICATION (launcher),
                                     g_steal_pointer (&option_group));
 }
+
+static gboolean
+option_entry_parse_permissions (const char          *option G_GNUC_UNUSED,
+                                const char          *value,
+                                CogLauncher         *launcher,
+                                GError             **error)
+{
+
+    if (g_strcmp0 (value, "all") && g_strcmp0 (value, "none")) {
+        g_set_error (error,
+                     G_OPTION_ERROR,
+                     G_OPTION_ERROR_BAD_VALUE,
+                     "Invalid permission value '%s' (allowed values: ['none', 'all'])",
+                     value);
+        return FALSE;
+    }
+
+    launcher->permissions = g_strdup (value);
```

Comment:
Going with the suggestions above, for example if the value is a boolean, this would be just:

```c
    launcher->permissions = g_strcmp0 (value, "all") == 0;
```

We would avoid the string copy, too :wink: